providers/s3/s3.go

```diff
@@ -227,12 +228,18 @@ func NewBucketWithConfig(logger log.Logger, config Config, component string) (*B
 			},
 		})}
 	} else {
+		hedgedClient, err := hedgedhttp.New(hedgedhttp.Config{
+			Transport: http.DefaultTransport,
+		})
+		if err != nil {
+			return nil, err
+		}
 		chain = []credentials.Provider{
 			wrapCredentialsProvider(&credentials.EnvAWS{}),
 			wrapCredentialsProvider(&credentials.FileAWSCredentials{}),
 			wrapCredentialsProvider(&credentials.IAM{
 				Client: &http.Client{
```

Comment:
The Transport need to come from this function here: https://github.com/thanos-io/objstore/blob/main/client/factory.go#L65